src/main/java/org/disq_bio/disq/impl/formats/sam/SamFormat.java

```diff
@@ -54,4 +57,17 @@ public static SamFormat fromPath(String path) {
     }
     return null;
   }
+
+  public AbstractSamSource createAbstractSamSource(FileSystemWrapper fileSystemWrapper) {
+    switch (this) {
```

Comment:
The same as before: instead of a switch with `this`, the way will be to have a `Function<FileSystemWrapper, AbstractSamSource>` instead, which makes it cleaner (and does not require checking for equality).